zrml/prediction-markets/src/lib.rs

```diff
@@ -422,7 +448,7 @@ mod pallet {
                 ScoringRule::CPMM,
             )?
             .actual_weight
-            .unwrap_or_else(T::WeightInfo::create_market);
+            .unwrap_or_else(|| T::WeightInfo::create_market(CacheSize::get()));
```

Comment:
With `create_market` we call an extrinsic [here](https://github.com/zeitgeistpm/zeitgeist/blob/0176500e49a1ba079cbdd6bc18cfff943b71f18c/zrml/prediction-markets/src/lib.rs#L499-L607). Unfortunately it's not guaranteed if there is some `actual_weight`. To avoid this `boilerplate syntax` I would probably follow another approach like Parity does. To avoid this, one can use a function which returns a `DispatchResult` and this function is called by the extrinsics `create_market` and `create_cpmm_market_and_deploy_assets`.

Reply:
Not sure I completely understand. Why not just put the actual weight in the post info?

Reply:
https://github.com/paritytech/substrate/blob/409167ef427e1b1b32a0cb9c05117e8fe73a7b1a/frame/transaction-payment/src/lib.rs#L1070-L1080

The actual weight can always be `None` in `DispatchResultWithPostInfo` with my understanding.

Quick check from yesterday: We don't use `calculate_actual_weight` here.

Reply:
That's what I'm suggesting. Always return `Some` weight in the post info.

Reply:
The `actual_weight` is `None` for early return errors. But because we already defined the worst case in the header of `create_cpmm_market_and_deploy_assets` above, it already takes the worst case for an early return error. That's why we could use the `ok_or` syntax. I would probably extend the user experience further. For an early return error in `create_market` we could return `Some(create_market_worst_case_weight)` instead of the expensive default weight of `create_cpmm_market_and_deploy_assets`.

Reply:
According to [this](https://github.com/paritytech/substrate/blob/a47f200eebeb88a5bde6f1ed2be9728b82536dde/frame/support/src/dispatch.rs#L204-L286) the `None` is specified when it's the default case (specified from `#[pallet::weight(...)]`). Because the extrinsic `create_market` returns with `?` and then the `actual_weight` is queried, we can expect, that `actual_weight` is always `Some` weight (a successful Ok(Some(...)) case). 

So, I love your suggestion now.

Reply:
3eba61d31d92b1007441c8dfe6ac774b55cc7508